CXSPA-752 [BOPIS] Implement Pickup functionality on PDP

SVendittelli opened the pull request.

Commit:
feat: add cart wrapper to pickup in store

Commit:
test: improve PickupOptionsComponent tests

Document the PickupOptionsComponent and extend unit tests to cover the
template.

CXSPA-752

Commit:
docs: improve bopis documentation

CXSPA-752

Commit:
refactor: remove unused dependency

CXSPA-752

Commit:
feat: move bopis behind env feature flag

CXSPA-752

Commit:
fix: correct in stock checks

CXSPA-752

Commit:
test: new out of stock logic

CXSPA-752

Commit:
test: bopis active cart service

JedidiahAwuku commented on feature-libs/pickup-in-store/components/presentational/store/store.component.html:
```diff
@@ -42,19 +42,23 @@
       <cx-icon
         class="cx-store-stock-icon"
-        *ngIf="!!storeDetails?.stockInfo?.stockLevel"
+        *ngIf="isInStock()"
```
This should be a variable instead of function call.

SVendittelli replied:
Done

JedidiahAwuku commented on feature-libs/pickup-in-store/components/presentational/store/store.component.ts:
```diff
@@ -25,4 +25,11 @@ export class StoreComponent {
     this.openHoursOpen = !this.openHoursOpen;
   }
+
+  isInStock(): boolean {
```
This should be replaced with a variable.

SVendittelli replied:
Done

JedidiahAwuku commented on projects/storefrontapp/src/environments/environment.ts:
```diff
@@ -23,4 +23,5 @@ export const environment: Environment = {
   cpq: buildProcess.env.CX_CPQ ?? false,
   digitalPayments: buildProcess.env.CX_DIGITAL_PAYMENTS ?? false,
   epdVisualization: buildProcess.env.CX_EPD_VISUALIZATION ?? false,
+  pickupInStore: buildProcess.env.CX_PICKUP_IN_STORE ?? false,
```
Default to true but leave a todo to change to false.

SVendittelli replied:
Done

Commit:
fix: remove function calls from template

To prevent repeated execution of functions on changes, extract the out
of stock logic to utility and set variable on init.

Additionally default bopis to be on for development purposes.

Commit:
fix: remove circular dependency

Commit:
test: correct cypress expectations

As low stock is not available for pickup, we need to change the store we
use in our cypress tests.
